File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  // type your code here

  if (num < 1) return 0;

  var a, b, result;
  a = 0;
  b = 1;
  result = b;
  for (let i = 1; i < num; i++) {
    result = a + b;
    a = b;
    b = result;
  }
  return result;
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  # type your code in here

  $a=0
  $b=1
  $result

  return $a if num === 0
  return $b if num === 1
  
  (num-1).times do
    $result=$a+$b
    $a=$b
    $b=$result
  end
  
  $result
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!

end

# Please add your pseudocode to this file
# And a written explanation of your solution
